sample/trace_key_test.go

```diff
@@ -160,3 +162,341 @@ func createTestTrace(t *testing.T, spans []testSpan) *types.Trace {
 
 	return trace
 }
+
+func TestDistinctValue_AddAsString(t *testing.T) {
+	fields := []string{"field1", "field2"}
+	dv := &distinctValue{
+		buf: make([]byte, 0, 1024),
+	}
+	dv.init(fields, 10)
+
+	tests := []struct {
+		name        string
+		value       any
+		fieldIdx    int
+		wantSuccess bool
+		wantCount   int
+	}{
+		{
+			name:        "Add string",
+			value:       "test-value",
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   1,
+		},
+		{
+			name:        "Add integer",
+			value:       42,
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   2,
+		},
+		{
+			name:        "Add float",
+			value:       3.14,
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   3,
+		},
+		{
+			name:        "Add boolean",
+			value:       true,
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   4,
+		},
+		{
+			name:        "Add to second field",
+			value:       "second-field",
+			fieldIdx:    1,
+			wantSuccess: true,
+			wantCount:   5,
+		},
+		{
+			name:        "Add duplicate string",
+			value:       "test-value",
+			fieldIdx:    0,
+			wantSuccess: false, // Already exists
+			wantCount:   5,     // Unchanged
+		},
+		{
+			name:        "Add nil",
+			value:       nil,
+			fieldIdx:    0,
+			wantSuccess: false,
+			wantCount:   5, // Unchanged
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := dv.AddAsString(tt.value, tt.fieldIdx)
+			require.Equal(t, tt.wantSuccess, got)
+			require.Equal(t, tt.wantCount, dv.totalUniqueCount)
+		})
+	}
+}
+
+func TestDistinctValue_MaxLimit(t *testing.T) {
+	maxValues := 3
+	dv := &distinctValue{
+		buf: make([]byte, 0, 1024),
+	}
+	dv.init([]string{"field1"}, maxValues)
+
+	// Add up to the limit
+	for i := 0; i < maxValues-1; i++ {
+		require.True(t, dv.AddAsString(i, 0), "Expected success adding value %d, got failure", i)
+	}
+
+	// This should fail as we've reached the limit
+	require.False(t, dv.AddAsString("one-too-many", 0), "Expected failure when adding beyond max limit, got success")
+
+	require.Equal(t, maxValues, dv.totalUniqueCount)
+}
+
+func TestDistinctValue_Values(t *testing.T) {
+	dv := &distinctValue{
+		buf: make([]byte, 0, 1024),
+	}
+	dv.init([]string{"field1", "field2", "empty-field"}, 10)
+
+	// Add some mixed values
+	dv.AddAsString("banana", 0)
+	dv.AddAsString("apple", 0)
+	dv.AddAsString("cherry", 0)
+	dv.AddAsString(42, 1)
+	dv.AddAsString(true, 1)
+
+	t.Run("Values sorted", func(t *testing.T) {
+		values := dv.Values(0)
+		expected := []string{"apple", "banana", "cherry"}
+		require.EqualValues(t, expected, values)
+	})
+
+	t.Run("Type conversion", func(t *testing.T) {
+		values := dv.Values(1)
+		require.Len(t, values, 2)
+		require.Contains(t, values, "42")
+		require.Contains(t, values, "true")
+	})
```

Comment:
Since `values` is sorted you should be able to compare equality of the whole slice, which IMO is a more robust test.